docs/site/Loopback-component-authentication.md

```diff
@@ -0,0 +1,600 @@
+---
+lang: en
+title: 'Authentication'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Loopback-component-authentication.html
+---
+
+## Overview
+
+Security is of paramount importance when developing a web or mobile application
+and usually consists of two distinct pieces :
+
+- Authentication
+- Authorization
+
+`Authentication` is a process of verifying someone's identity before a protected
+endpoint is accessed.
+
+`Authorization` is a process of verifying the actions that an authenticated user
+is allowed to perform on a protected endpoint.
+
+This document describes the details of the LoopBack 4 `Authentication`
+component.
+
+{% include note.html content="
+For a description of an `Authorization` process, please see <a href='https://loopback.io/doc/en/lb4/Loopback-component-authorization.html'>Authorization</a>.
+" %}
+
+## Authentication Component
+
+To utilize `authentication` in an application `application.ts`, you must load
+the authentication component named `AuthenticationComponent`.
```

Comment:
Apologize that I'm not familiar with this feature. So the name has to be `AuthenticationComponent`? That means we cannot have more than one Authentication component?

Reply:
@dhmlau 

> Apologize that I'm not familiar with this feature. So the name has to be `AuthenticationComponent`? That means we cannot have more than one Authentication component?

So most of the logic of @loopback/authentication ( the authentication metadata provider, the authentication action provider, and the authentication strategy provider ) is available in a component named `AuthenticationComponent`, and must be loaded by an application.  

The user needs to create and register their own custom authentication strategies (user can have many of these), and implement their own token service or user service etc, and create a custom sequence that places the authentication action in a specific place.

But the core `authentication framework logic` is stored in 1 component.

Reply:
@dhmlau ^^ :)